### PR TITLE
Added child container for Region in dynamic grid

### PR DESCRIPTION
**Pull request by msaikaushik1992**
### Summary

This ensures the content inside a region of dynamic grid resizes based on its parent region and not more.

### Additional Details

Closes https://github.com/cerner/terra-core/issues/992

### Deployment Links
https://terra-core-deployed-pr-1043.herokuapp.com/static/#/site/dynamic-grid
https://terra-core-deployed-pr-1043.herokuapp.com/static/#/tests/dynamic-grid-tests

### CODE REVIEW

File: packages/terra-dynamic-grid/src/Region.jsx [hunk 1]
```diff
@@ -104,7 +104,13 @@ const Region = ({
   });
 
   const regionClasses = classNames(customProps.className, css(stylesheet.region));
-  return (<div {...customProps} className={regionClasses}>{children}</div>);
+  return (
+    <div {...customProps} className={regionClasses}>
+      <div className="region-child-container">
```
Comment (mhemesath) [inline packages/terra-dynamic-grid/src/Region.jsx]: this class should be localized via `cx('region-child-container')`

File: packages/terra-dynamic-grid/src/Region.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+:local {
+  .region-child-container {
```
Comment (mhemesath) [inline packages/terra-dynamic-grid/src/Region.scss]: can we add  a comment to why this is needed here

File: packages/terra-dynamic-grid/src/Region.jsx (hunk 1 above)
Comment (msaikaushik1992) [reply packages/terra-dynamic-grid/src/Region.jsx]: Updated https://github.com/cerner/terra-core/pull/1043/commits/1d10ad3a4b1777c2555a6c0bee86be76e4f955f3#diff-e9aa781e0a2995c2193ef9eb6a078ecfR5

File: packages/terra-dynamic-grid/src/Region.scss (hunk 2 above)
Comment (msaikaushik1992) [reply packages/terra-dynamic-grid/src/Region.scss]: Updated https://github.com/cerner/terra-core/pull/1043/commits/1d10ad3a4b1777c2555a6c0bee86be76e4f955f3#diff-765a8fc225f42601c30ea66ab554cb91R1

**Comment by mmalaker** [comment]
Functional verification completed. The dynamic grid examples now display and function correctly.

**Pull request merged**